sdk/agent/bufferedagent/agent.go

```diff
@@ -78,66 +87,55 @@ func (a *Agent) Open() error {
 
 // Payment queues a payment which will be paid in the next settlement. The
 // identifier for the settlement is returned.
-func (a *Agent) Payment(paymentAmount int64) (settlementID string) {
+func (a *Agent) Payment(paymentAmount int64) (queueID string, err error) {
 	a.mu.Lock()
 	defer a.mu.Unlock()
+	if a.maxQueueSize != 0 && len(a.queue) == a.maxQueueSize {
+		return "", ErrQueueFull
+	}
+	if paymentAmount > math.MaxInt64-a.queueTotalAmount {
+		return "", ErrQueueFull
```

Comment:
💡  this is technically a different error right? Might be deceiving if a test hits this limit and this error gets returned. What about a custom error like `max payment amount reached`